services/galley/src/Galley/API/Action.hs

```diff
@@ -634,7 +666,26 @@ updateLocalConversationUnchecked lconv qusr con action = do
       conv = tUnqualified lconv
 
   -- retrieve member
-  self <- noteS @'ConvNotFound $ getConvMember lconv conv qusr
+  self <-
+    if (cnvmType . convMetadata . tUnqualified $ lconv) == GlobalTeamConv
+      then
+        -- TODO(elland): address this problem
+        pure . Left $
```

Comment:
(*Referring to this whole function*)

This is not so terrible, but the GTC querying bits could be moved to a different function to cleanly separate the special logic for GTCs from the rest of the logic.

Of course, the "clean" haskelly solution would be to make `Conversation` a sum type encompassing the two kinds of conversations (i.e. regular or GTC). Given the unfriendliness of record types in Haskell, I understand you might have some reservations about going this route, but maybe it's worth considering.

Reply:
> make Conversation a sum type 

I'm all for it but this PR is large enough as it is. Should we create a follow up?